templates/consul_systemd.service.j2

```diff
@@ -23,6 +23,7 @@ Environment=CONSUL_UI_LEGACY=true
 {% endif %}
 ExecStartPre=-/bin/mkdir -m 0750 -p {{ consul_run_path }}
 ExecStartPre=/bin/chown -R {{ consul_user }}:{{ consul_group }} {{ consul_run_path }}
+ExecStartPre={{ consul_binary }} validate -quiet {{ consul_config_path }}/config.json {{ consul_configd_path }}
```

Comment:
Two questions:

1. Why do we need to validate on service startup? We already do it when configuring the task.
2. Why only do this for systemd? There are several more service manager files in `templates` (namely, for SysV init, macOS launchctl, BSD, etc. This change should be applied to all of them, or not at all.

I would suggest dropping this change. I think it's enough to validate the config file after configuring it in `tasks/config.yml`.

Reply:
@abarbare I made it consistent using `consul_binary` on the other call, so the binary name isn't hardcoded anywhere.

@nre-ableton 
1.  It ensure any external configuration change (manual change for exemple) will be validated. The validate gives you a non 0 return code and system will show you the error right away, instead of failing silently (a systemctl restart consul won't tell you anything about any issue). I thinks it's a nice and free addition, but not mandatory. If you really prefer to remove it I'll do it. The validation during the playbook run is the most important part.  
2. I only have systemd based system to test it, but I can probably use molecule to validated it ? (never done any automated testing with ansible). 

If you are willing to keep the change I can look into adding and testing the change into other service manager init script.

Reply:
Sounds like a plan @nre-ableton! Thanks for digging into it, I'll work on it, glad to contribute :)

Reply:
Updated @nre-ableton :+1: